app/models/grda_warehouse/hud/project.rb

```diff
@@ -334,6 +335,33 @@ def serves_children?
       SQL
     end
 
+    private_class_method def self.has_access_to_project_through_coc_codes(user, q, qc)
+      return '(1=0)' unless user.coc_codes.any?
+
+      project_coc_table = GrdaWarehouse::Hud::ProjectCoc.quoted_table_name
```

Comment:
See `GrdaWarehouse::ServiceHistoryEnrollment.with_service_between` for how to accomplish this with Arel.

Reply:
I'm not sure what you have in mind.